trade-finance-manager-ui/server/controllers/case/activity/index.test.js

```diff
@@ -18,354 +18,341 @@ const session = {
   },
 };
 
-describe('GET activity', () => {
-  describe('when deal exists', () => {
-    const mockDeal = {
-      _id: '61f6ac5b02fade01b1e8efef',
-      dealSnapshot: {
+describe('activity controller', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('GET activity', () => {
+    describe('when deal exists', () => {
+      const mockDeal = {
         _id: '61f6ac5b02fade01b1e8efef',
-        submissionDetails: {
-          supplierName: 'test supplier',
+        dealSnapshot: {
+          _id: '61f6ac5b02fade01b1e8efef',
+          dealType: DEAL_TYPE.GEF,
+          submissionDetails: {
+            supplierName: 'test supplier',
+          },
+          ukefDealId: 'ukefDealId',
         },
-      },
-      tfm: {
-        activities: [],
-      },
-    };
-
-    beforeEach(() => {
-      api.getDeal = () => Promise.resolve(mockDeal);
-      api.getAmendmentsByDealId = () => Promise.resolve({ data: [] });
-    });
-
-    it('should render template with data', async () => {
-      const req = {
-        params: {
-          _id: mockDeal._id,
+        tfm: {
+          activities: [],
         },
-        session,
       };
 
-      const activities = mapActivities(mockDeal.tfm.activities);
-
-      await activityController.getActivity(req, res);
-      expect(res.render).toHaveBeenCalledWith('case/activity/activity.njk', {
-        activePrimaryNavigation: 'manage work',
-        activeSubNavigation: 'activity',
-        deal: mockDeal.dealSnapshot,
-        tfm: mockDeal.tfm,
-        dealId: mockDeal.dealSnapshot._id,
-        user: session.user,
-        selectedActivityFilter: CONSTANTS.ACTIVITIES.ACTIVITY_FILTERS.ALL,
-        activities,
-        amendmentsInProgress: [],
-        hasAmendmentInProgress: false,
+      beforeEach(() => {
+        api.getDeal = () => Promise.resolve(mockDeal);
+        api.getAmendmentsByDealId = () => Promise.resolve({ data: [] });
+        api.getDealCancellation = () =>
+          Promise.resolve({
+            effectiveFrom: new Date().valueOf(),
+            bankRequestDate: new Date().valueOf(),
+            reason: 'a reason',
+            status: TFM_DEAL_CANCELLATION_STATUS.COMPLETED,
+          });
       });
-    });
-  });
 
-  describe('when deal does NOT exist', () => {
-    beforeEach(() => {
-      api.getDeal = () => Promise.resolve();
-      api.getAmendmentsByDealId = () => Promise.resolve({ data: [] });
+      it('should render template with data', async () => {
+        const mockSuccessMessage = 'mockSuccessMessage';
+
+        const req = {
+          params: {
+            _id: mockDeal._id,
+          },
+          session,
+          flash: jest.fn(() => [mockSuccessMessage]),
```

Comment:
Instead of this and `successMessage: mockSuccessMessage` below, I think the tests should simply call `getDealSuccessBannerMessage`

Reply:
Sorry, didn't realise it was async. I would prefer to mock `getDealSuccessBannerMessage` in the test so it's more realistic, but maybe overkill. What do you think?

Reply:
Misunderstood this comment! Have updated to mock all the async calls of `getDealSuccessBannerMessage`